### PR TITLE
feat: braze router batching update

### PR DESCRIPTION
**Pull request by aashishmalik**
## Description of the change

> Updated router batching in braze to support subscription calls

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/v0/destinations/braze/util.js [hunk 1]
```diff
@@ -374,7 +391,33 @@ const processBatch = (transformedEvents) => {
       headers,
     });
   }
-  const finalResponse = [];
+
+  for (let i = 0; i < subscriptionArrayChunks.length; i += 1) {
```
Comment (koladilip) [inline src/v0/destinations/braze/util.js:395]: following code blocks are very similar, make it function and reuse for both of them

Comment (sanpj2292) [reply src/v0/destinations/braze/util.js:395]: ➕

File: src/v0/destinations/braze/config.js [hunk 2]
```diff
@@ -18,7 +18,7 @@ function getTrackEndPoint(endPoint) {
 }
 
 function getSubscriptionGroupEndPoint(endPoint) {
-  return `${endPoint}/subscription/status/set`;
+  return `${endPoint}/v2/subscription/status/set`;
```
Comment (utsabc) [inline src/v0/destinations/braze/config.js:21]: Is this api compatible with the previous one?

File: src/v0/destinations/braze/braze.util.test.js [hunk 3]
```diff
@@ -765,6 +768,11 @@ describe('processBatch', () => {
     expect(result[0].batchedRequest[1].body.JSON.attributes.length).toBe(25); // Second batch contains remaining 25 attributes
     expect(result[0].batchedRequest[1].body.JSON.events.length).toBe(25); // Second batch contains remaining 25 events
     expect(result[0].batchedRequest[1].body.JSON.purchases.length).toBe(25); // Second batch contains remaining 25 purchases
+    expect(result[0].batchedRequest[2].body.JSON.subscription_groups.length).toBe(50); // First batch contains 50 subscription group
+    expect(result[0].batchedRequest[3].body.JSON.subscription_groups.length).toBe(50); // First batch contains 25 subscription group
+    expect(result[0].batchedRequest[4].body.JSON.merge_updates.length).toBe(50); // First batch contains 50 merge_updates
+    expect(result[0].batchedRequest[5].body.JSON.merge_updates.length).toBe(50); // First batch contains 25 merge_updates
```
Comment (utsabc) [inline src/v0/destinations/braze/braze.util.test.js:774]:
```suggestion
    expect(result[0].batchedRequest[3].body.JSON.subscription_groups.length).toBe(50); // First batch contains 50 subscription group
    expect(result[0].batchedRequest[4].body.JSON.merge_updates.length).toBe(50); // First batch contains 50 merge_updates
    expect(result[0].batchedRequest[5].body.JSON.merge_updates.length).toBe(50); // First batch contains 50 merge_updates
```

Comment (aashishmalik) [reply src/v0/destinations/braze/braze.util.test.js]: done

File: src/v0/destinations/braze/config.js (hunk 2 above)
Comment (aashishmalik) [reply src/v0/destinations/braze/config.js:21]: yes, updated transformer code to work with previous payload from user and transforming that code as per this api

**Review by utsabc — DISMISSED** [review]

File: src/v0/destinations/braze/util.js [hunk 4]
```diff
@@ -317,18 +319,51 @@ const processDeduplication = (userStore, payload, destinationId) => {
...
+  const headers = {
+    'Content-Type': JSON_MIME_TYPE,
+    Accept: JSON_MIME_TYPE,
+    Authorization: `Bearer ${destination.Config.restApiKey}`,
+  };
+
+  for (let chunk of arrayChunks) {
+    const response = defaultRequestConfig();
+    if (type === 'merge') {
+      response.endpoint = getAliasMergeEndPoint(getEndpointFromConfig(destination));
```
Comment (koladilip) [inline src/v0/destinations/braze/util.js:333]: both blocks are very similar try to write reusable function

File: src/v0/destinations/braze/util.js [hunk 5]
```diff
@@ -338,24 +373,37 @@ const processBatch = (transformedEvents) => {
...
+      }
+
+      if (Array.isArray(merge_updates)) {
+        mergeUsersArray.push(...merge_updates);
+      }
+
       successMetadata.push(...transformedEvent.metadata);
     }
   }
   const attributeArrayChunks = _.chunk(attributesArray, 75);
```
Comment (koladilip) [inline src/v0/destinations/braze/util.js:388]: 75, 50 these should be constants defined in config.js

Comment (aashishmalik) [reply src/v0/destinations/braze/util.js:388]: updated

File: src/v0/destinations/braze/util.js (hunk 4 above)
Comment (aashishmalik) [reply src/v0/destinations/braze/util.js:336]: both block has different endpoint and payload `key` name ... other common code is separated out

**Review by utsabc — APPROVED** [review]

**Pull request merged**